programs/wen_transfer_guard/src/instructions/initialize.rs

```diff
@@ -0,0 +1,54 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{token_2022::spl_token_2022::ID as TOKEN_2022_PROGRAM_ID, token_interface::Mint};
+use spl_tlv_account_resolution::{account::ExtraAccountMeta, state::ExtraAccountMetaList};
+use spl_transfer_hook_interface::{error::TransferHookError, instruction::ExecuteInstruction};
+
+use crate::{AnchorExtraAccountMeta, EXTRA_ACCOUNT_METAS};
+
+#[derive(Accounts)]
+#[instruction(metas: Vec<AnchorExtraAccountMeta>)]
+pub struct Initialize<'info> {
+    /// CHECK: This account's data is a buffer of TLV data
+    #[account(
+        init,
+        space = ExtraAccountMetaList::size_of(metas.len()).unwrap(),
+        // space = 8 + 4 + 2 * 35,
+        seeds = [EXTRA_ACCOUNT_METAS.as_ref(), mint.key().as_ref()],
+        bump,
+        payer = payer,
+    )]
+    pub extra_metas_account: UncheckedAccount<'info>,
+
+    #[account(mint::token_program = TOKEN_2022_PROGRAM_ID)]
+    pub mint: Box<InterfaceAccount<'info, Mint>>,
+
+    #[account(mut)]
+    pub mint_authority: Signer<'info>,
```

Comment:
only issue I see with using the mint authority is that in practice we will want there to not be any more MintAuth on these tokens after the initial minting phase (to ensure supply stays <=1) so you would not be able to add the transfer hook as a separate thing after the fact. I wonder if there's a different authority (metadata auth?) that could be better long term.

Reply:
Thanks we are exploring alternative method.

Reply:
Using Transfer Hook authority right now